chore(Jeninsfile) Switch Maven commands to infra.runWithMaven

dduportal opened the pull request:
The goal is to benefits from infra's JDK and maven setups per agent kind.

It avoid infamous errors `The JAVA_HOME environment variable is not defined correctly,
` in the ATH and benefits the work from https://github.com/jenkins-infra/pipeline-library/pull/231 abd https://github.com/jenkins-infra/pipeline-library/pull/232 (not downloading JDK on each agent)

Related to #5841 which "ath" step failed.

- First manual test (used "replay" with the PR's pipeline as I'm not maintainer,but admin on ci.jenkins.io):  https://ci.jenkins.io/job/Core/job/jenkins/job/PR-5843/11/

### Proposed changelog entries

N.A.

<!-- Comment: 
The changelogs will be integrated by the core maintainers after the merge.  See the changelog examples here: https://www.jenkins.io/changelog/ -->

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

<!-- For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc) -->

### Desired reviewers

@jglick @timja 

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/code-reviewers
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

jglick reviewed (changes requested):
Accidental deletion of important `git diff` call.

jglick reviewed (approved):
Looks like it should work.

jglick reviewed (commented):
As a rule, I really dislike `runMaven` and think it should be deprecated. Prefer `runWithMaven` + `sh 'mvn …'` for transparency. Would that suffice to solve the issues mentioned in the description?

dduportal committed:
chore(Jeninsfile) Switch Maven commands to infra.runWithMaven

The goal is to benefit from the the shared library jenkins-infra/pipeline-library
that manages agents and tools.

Signed-off-by: Damien Duportal <damien.duportal@gmail.com>
